cirq-core/cirq/value/measurement_key.py

```diff
@@ -41,7 +41,7 @@ class MeasurementKey:
     path: Tuple[str, ...] = dataclasses.field(default_factory=tuple)
 
     def __post_init__(self):
-        if not self.name:
+        if self.name is None:
```

Comment:
`name` has annotated type `str`, not `Optional[str]`. How can it be `None`? Just from the code it looks like this should be an isinstance check, but I know you intend None to be some kind of placeholder instead.